Mathlib/Analysis/InnerProductSpace/Basic.lean

```diff
@@ -231,18 +259,19 @@ theorem inner_zero_left (x : F) : ⟪0, x⟫ = 0 := by
 theorem inner_zero_right (x : F) : ⟪x, 0⟫ = 0 := by
   rw [← inner_conj_symm, inner_zero_left]; simp only [RingHom.map_zero]
 
-theorem inner_self_eq_zero {x : F} : ⟪x, x⟫ = 0 ↔ x = 0 :=
-  ⟨c.definite _, by
-    rintro rfl
-    exact inner_zero_left _⟩
+theorem inner_self_of_eq_zero {x : F} : x = 0 → ⟪x, x⟫ = 0 := by
+    intro h
+    rw [h]
+    exact inner_zero_left _
 
-theorem normSq_eq_zero {x : F} : normSqF x = 0 ↔ x = 0 :=
-  Iff.trans
-    (by simp only [normSq, ext_iff, map_zero, inner_self_im, eq_self_iff_true, and_true_iff])
-    (@inner_self_eq_zero 𝕜 _ _ _ _ _ x)
+theorem normSq_eq_zero_of_eq_zero {x : F} : x = 0 → normSqF x = 0 := by
+  intro h
+  rw [normSq]
+  rw [inner_self_of_eq_zero h]
+  exact RCLike.zero_re'
 
-theorem inner_self_ne_zero {x : F} : ⟪x, x⟫ ≠ 0 ↔ x ≠ 0 :=
-  inner_self_eq_zero.not
+theorem ne_zero_of_inner_self_ne_zero {x : F} : ⟪x, x⟫ ≠ 0 → x ≠ 0 := by
+  exact fun P Q => P (inner_self_of_eq_zero Q)
 
```

Comment:
This is just the contrapositive of the existing result, which we can obtain with `mt` (modus tollens)
```suggestion
theorem ne_zero_of_inner_self_ne_zero {x : F} : ⟪x, x⟫ ≠ 0 → x ≠ 0 :=
  mt inner_self_of_eq_zero
```